test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -196,13 +196,13 @@ private void HandleHttpRequests()
                 try
                 {
                     var ctx = _listener.GetContext();
-                    var rawSpans = MessagePackSerializer.Deserialize<dynamic>(ctx.Request.InputStream);
-                    var spans = ToSpans(rawSpans);
-
                     lock (this)
                     {
                         // we only need to lock when replacing the span collection,
                         // not when reading it because it is immutable
+                        // We also need to lock when deserializing the streams as well, in case there are multiple concurrent requests coming in
```

Comment:
Really? Each request gets its own context with a distinct `Request` and `InputStream`. They are not shared between requests.